tests/src/Kernel/Entity/OgMembershipTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\og\Kernel\Entity\OgMembershipTest.
+ */
+
+namespace Drupal\Tests\og\Kernel\Entity;
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+use Drupal\user\Entity\User;
+
+/**
+ * Tests the OgMembership entity.
+ *
+ * @group og
+ * @coversDefaultClass \Drupal\og\Entity\OgMembership
+ */
+class OgMembershipTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'entity_test',
+    'field',
+    'og',
+    'system',
+    'user',
+  ];
+
+  /**
+   * Test group.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  protected $group;
+
+  /**
+   * Test users.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $user;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installConfig(['og']);
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('entity_test');
+    $this->installEntitySchema('user');
+    $this->installSchema('system', 'sequences');
+
+    // Create a bundle and add as a group
+    $group = EntityTest::create([
+      'type' => Unicode::strtolower($this->randomMachineName()),
+      'name' => $this->randomString(),
+    ]);
+
+    $group->save();
+    $this->group = $group;
+
+    // Add that as a group.
+    Og::groupManager()->addGroup('entity_test', $group->id());
+
+    // Create test user.
+    $user = User::create(['name' => $this->randomString()]);
+    $user->save();
+
+    $this->user = $user;
+  }
+
+  /**
+   * Tests getting abd setting users on OgMemberships.
+   *
+   * @covers ::getUser
+   * @covers ::setUser
+   */
+  public function testGetSetUser() {
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->user)
+      ->setEntityId($this->group->id())
+      ->setGroupEntityType($this->group->getEntityTypeId())
+      ->save();
+
+    // Check the user is returned.
+    $this->assertEquals($this->user->id(), $membership->getUser()->id());
+
+    // And after re-loading.
+    $membership = Og::membershipStorage()->loadUnchanged($membership->id());
+
+    $this->assertEquals($this->user->id(), $membership->getUser()->id());
+  }
+
+  /**
+   * Tests exceptions are thrown when trying to save a membership with no, or
+   * anonymous user.
+   *
+   * @covers ::getUser
+   * @dataProvider providerTestGetSetUserException
+   * @expectedException \Drupal\Core\Entity\EntityStorageException
```

Comment:
Oh, I see! Yeah, this is awesome! You can also set it in code from the test with `$this->setExpectedException()`.